File: fixtures/scale/dart/large/lib/ciphers/fletcher/fletcher_suite.dart
import '../../core/cipher_suite.dart';
import '../../core/hash_function.dart';
import '../../core/suite_result.dart';
import '../../verify/determinism.dart';
import 'fletcher_hash.dart';
import 'fletcher_vectors.dart';

class FletcherSuite implements CipherSuite {
  @override
  String name() => 'fletcher';

  @override
  String category() => 'hash';

  @override
  SuiteResult run() {
    final HashFunction hash = FletcherHash();
    final samples = fletcherSamples();
    var passed = 0;
    var failed = 0;
    final start = DateTime.now();
    for (final sample in samples) {
      if (stableDigest(hash, sample)) passed++; else failed++;
    }
    for (final other in samples) {
      if (distinctDigests(hash, samples[0], other)) passed++; else failed++;
    }
    final elapsed = DateTime.now().difference(start).inMicroseconds;
    return SuiteResult(name(), passed, failed, elapsed);
  }
}

File: fixtures/scale/kotlin/large/src/cryptobench/ciphers/fletcher/FletcherSuite.kt
package cryptobench.ciphers.fletcher

import cryptobench.core.CipherSuite
import cryptobench.core.HashFunction
import cryptobench.core.SuiteResult
import cryptobench.verify.distinctDigests
import cryptobench.verify.stableDigest

class FletcherSuite : CipherSuite {
    override fun name(): String = "fletcher"

    override fun category(): String = "hash"

    override fun run(): SuiteResult {
        val hash: HashFunction = FletcherHash()
        val samples = FletcherVectors.samples()
        var passed = 0
        var failed = 0
        val start = System.nanoTime()
        for (sample in samples) {
            if (stableDigest(hash, sample)) passed++ else failed++
        }
        for (other in samples) {
            if (distinctDigests(hash, samples[0], other)) passed++ else failed++
        }
        return SuiteResult(name(), passed, failed, System.nanoTime() - start)
    }
}
